### ISSUE TITLE
A possible locking issue

### ISSUE THREAD

**Issue by anson-lo** [memcached/memcached#885]
Hi, should the lock be released after Line 863 to protect `l->dropped++` from races?
https://github.com/memcached/memcached/blob/894e4c16b5162e25ce649cf8a57dfad250965ab0/logger.c#L858-L865

like this way
```
 pthread_mutex_lock(&l->mutex); 
 /* Request a maximum length of data to write to */ 
 e = (logentry *) bipbuf_request(buf, (sizeof(logentry) + reqlen)); 
 if (e == NULL) { 
     l->dropped++;
     pthread_mutex_unlock(&l->mutex); 
     return LOGGER_RET_NOSPACE; 
 } 
```

**Comment by dormando** [comment]
yup, I think that looks wrong. it's accessed under the lock during aggregation.

Thanks!

**Comment by dormando** [comment]
Fixed in 1.6.19
